dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/util/NeuralNetworkUtils.java

```diff
@@ -0,0 +1,258 @@
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters;
+import edu.snu.dolphin.dnn.layerparam.initializer.LayerParameterInitializer;
+import edu.snu.dolphin.dnn.layers.LayerBase;
+import edu.snu.dolphin.dnn.layers.LayerParameter;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Injector;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tang.formats.ConfigurationSerializer;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Utility class for neural network.
+ */
+public final class NeuralNetworkUtils {
+
+  private NeuralNetworkUtils() {
+  }
+
+  /**
+   * Delimiter that is used for distinguishing dimensions of shapes.
+   */
+  private static final String SHAPE_DELIMITER = ",";
+
+  /**
+   * Converts a list of integer for a shape to a string.
+   * @param dimensionList a list of integers for a shape.
+   * @return a string for a shape.
+   */
+  public static String shapeToString(final List<Integer> dimensionList) {
+    return StringUtils.join(dimensionList, SHAPE_DELIMITER);
+  }
+
+  /**
+   * Converts an array of {@code int} for a shape to a string.
+   * @param dimensions an array of {@code int}s for a shape.
+   * @return a string for a shape.
+   */
+  public static String shapeToString(final int[] dimensions) {
+    return shapeToString(Arrays.asList(ArrayUtils.toObject(dimensions)));
+  }
+
+  /**
+   * Converts a string for a shape to an array of integers.
+   * @param shapeString a string for a shape.
+   * @return an array of integers for a shape.
+   */
+  public static int[] shapeFromString(final String shapeString) {
+    final String[] inputShapeStrings = shapeString.split(SHAPE_DELIMITER);
+    final int[] inputShape = new int[inputShapeStrings.length];
+    for (int i = 0; i < inputShapeStrings.length; ++i) {
+      inputShape[i] = Integer.parseInt(inputShapeStrings[i]);
+    }
+    return inputShape;
+  }
+
+  public static int getShapeLength(final int[] shape) {
+    if (shape.length < 1) {
+      throw new IllegalArgumentException("the shape must have one or more dimensions");
+    }
+
+    int length = shape[0];
+    for (int i = 1; i < shape.length; ++i) {
+      length *= shape[i];
+    }
+
+    if (length > 0) {
+      return length;
+    } else {
+      throw new IllegalArgumentException("the length of the shape must be positive: " + length);
+    }
+  }
+
+  /**
+   * De-serializes a set of serialized layer configurations to an array of layer configurations.
+   * @param configurationSerializer a configuration serializer to deserialize the specified configuration set.
+   * @param serializedLayerConfSet a set of serialized layer configurations.
+   * @return an array of layer configurations.
+   */
+  public static Configuration[] deserializeLayerConfSetToArray(
```

Comment:
I get your point, but as a result we're iterating over the layer configurations twice instead of once per one `getInitialLayerParameters()` or `getLayerInstances()` call. I guess it wouldn't be a big problem since the number of layers isn't so big, though..